utils.go

```diff
@@ -44,49 +61,22 @@ func SelectHopHints(ctx context.Context, lnd *lndclient.LndServices,
 	// through GetChanInfo
 	chanInfoCache := make(map[uint64]*lndclient.ChannelEdge)
 
-	// skipCache is a simple cache which holds the indice of any
-	// channel we've added to final hopHints
+	// skipCache is a simple cache which holds the indices of any
+	// channel we've added to final hopHints.
 	skipCache := make(map[int]struct{})
 
 	hopHints := make([][]zpay32.HopHint, 0, numMaxHophints)
 
 	for i, channel := range openChannels {
+		channel := channel
+
 		// In this first pass, we'll ignore all channels in
 		// isolation that can't satisfy this payment.
 
-		// Retrieve extra info for each channel not available in
-		// listChannels
-		chanInfo, err := lnd.Client.GetChanInfo(ctx, channel.ChannelID)
-		if err != nil {
-			return nil, err
-		}
-
-		// Cache the GetChanInfo result since it might be useful
-		chanInfoCache[channel.ChannelID] = chanInfo
-
-		// Skip if channel can't forward payment
-		if channel.RemoteBalance < amtMSat {
-			log.Debugf(
-				"Skipping ChannelID: %v for hints as "+
-					"remote balance (%v sats) "+
-					"insufficient appears to be private",
-				channel.ChannelID, channel.RemoteBalance,
-			)
+		// Skip public or inactive channels.
+		if !chanCanBeHopHint(&channel) {
```

Comment:
Following on from comment above, would it make sense to also use `skipCache` here (since we're already using it for private nodes)? Saves us the duplicate check below.

Reply:
Would also perhaps add some debug logging, I'm sure somebody will wonder why an inactive private channel wasn't added so nice to be able to see that in the logs

Reply:
> Following on from comment above, would it make sense to also use skipCache here (since we're already using it for private nodes)? Saves us the duplicate check below.

Good idea, done.

> Would also perhaps add some debug logging, I'm sure somebody will wonder why an inactive private channel wasn't added so nice to be able to see that in the logs

PTAL